bots/mainbot/pilgrim.js

```diff
@@ -129,12 +148,20 @@ pilgrim.takeTurn = (self) => {
 							}
 						}
 						
-						// No nearby church. Build one
-						utilities.log(self, `Building a church nearby`);
-						pilgrim.mission = 'build';
+						utilities.log(self, `Karbonite: ${self.karbonite}, fuel: ${self.fuel}`)
+						if (self.fuel >= SPECS['UNITS'][SPECS['CHURCH']]['CONSTRUCTION_FUEL'] + 10
+								&& self.karbonite >= SPECS['UNITS'][SPECS['CHURCH']]['CONSTRUCTION_KARBONITE']) {
+
+							// No nearby church. Build one
+							utilities.log(self, `Building a church nearby`);
+							pilgrim.mission = 'build';
+						} else {
+							utilities.log(self, `Insufficient resources to build church. Returning home.`);
```

Comment:
Should we add functionality for churches to be build stuff? Though it might be hard to add them to the build queue

Reply:
We can certainly do that, (one of bovard's bots did this) but it should probably be a separate PR/branch as it will require a significant amount of code.